### PR TITLE
Add the gitlab bmulti library

### PR DESCRIPTION
**Pull request by correaa**
Proposing to add the B-Multi library to the live site. https://gitlab.com/correaa/boost-multi

Followed the steps as best as I could in https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/AddingALibrary.md#adding-a-new-library-to-the-live-site

Feedback on improving the hooks and the organizations of files in the repository is welcome.
The goal is to include the library by doing `#include<multi/array.hpp>`.

This PR is related to infra's PR https://github.com/compiler-explorer/infra/pull/745

### CODE REVIEW

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -1479,6 +1479,12 @@ libs.boost.versions.178.path=/opt/compiler-explorer/libs/boost_1_78_0
 libs.boost.versions.179.version=1.79.0
 libs.boost.versions.179.path=/opt/compiler-explorer/libs/boost_1_79_0
 
+libs.bmulti.name=B-Multi
+libs.bmulti.url=https://gitlab.com/correaa/boost-multi
+libs.bmulti.versions=trunk
+libs.bmulti.description=Modern C++ multidimensional arrays
+libs.bmulti.versions.trunk=/opt/compiler-explorer/libs/boost-multi/include
```
Comment (mattgodbolt) [inline etc/config/c++.amazon.properties:1486]:
```suggestion
libs.bmulti.versions.trunk.version=trunk
libs.bmulti.versions.trunk.path=/opt/compiler-explorer/libs/boost-multi/include
```

**Review by mattgodbolt — CHANGES_REQUESTED** [review]
these need to be `.path=<path>` and `.version=<name of version to show user>` Thanks!

**Review by mattgodbolt — APPROVED** [review]

**Comment by mattgodbolt** [comment]
Thanks! I'll do a site update tonight to pick this up :)

**Comment by correaa** [comment]
This is very exciting. Thank you.

Once all works I will step up to the next level and add it to the CUDA.godbolt if you agree. 
The library supports CUDA memory as well (https://gitlab.com/correaa/boost-multi#cuda-thrust)

**Comment by mattgodbolt** [comment]
This is now live!

**Comment by correaa** [comment]
Awesome.

How do I include it? If I put `#include<multi/array.hpp>` I get "fatal error: 'multi/array.hpp' file not found".

https://godbolt.org/z/qo4n3bedE

**Comment by correaa** [comment]
I might add it for Circle later as well (I made the library compatible with Circle recently, after some work).

**Comment by correaa** [comment]
Works now, thank you. How often does godbolt pulls the library from master? nightly I assume given the name of the keyword.

**Comment by correaa** [comment]
I noticed that MSVC compilers that used to be described with "WINE" now say "ex-WINE", and most libraries are not available for it anymore. What would be the correct way to integrate a header library, like this one, into these ex-WINE compilers?

**Pull request merged**